lib/screens/edit_product_screen.dart

```diff
@@ -107,6 +139,50 @@ class _EditProductScreenState extends State<EditProductScreen> {
     return value.isEmpty ? 'Error description exampler' : null;
   }
 
+  void onSavedImageUrl(String value) {
+    _editedProduct = Product(
+      title: _editedProduct.title,
+      price: _editedProduct.price,
+      description: _editedProduct.description,
+      imageUrl: value,
+      id: _editedProduct.id,
+      isFavorite: _editedProduct.isFavorite,
+    );
+  }
+
+  void onSavedDescription(String value) {
+    _editedProduct = Product(
+      title: _editedProduct.title,
+      price: _editedProduct.price,
+      description: value,
+      imageUrl: _editedProduct.imageUrl,
+      id: _editedProduct.id,
+      isFavorite: _editedProduct.isFavorite,
+    );
+  }
+
+  void onSavedPrice(String value) {
+    _editedProduct = Product(
+      title: _editedProduct.title,
+      price: double.parse(value),
+      description: _editedProduct.description,
+      imageUrl: _editedProduct.imageUrl,
+      id: _editedProduct.id,
+      isFavorite: _editedProduct.isFavorite,
+    );
+  }
+
+  void onSavedTitle(String value) {
+    _editedProduct = Product(
+      title: value,
+      price: _editedProduct.price,
+      description: _editedProduct.description,
+      imageUrl: _editedProduct.imageUrl,
+      id: _editedProduct.id,
+      isFavorite: _editedProduct.isFavorite,
+    );
+  }
```

Comment:
🤔

Reply:
🤕

Reply:
You are meant to explain these lines, just so you know

Reply:
These are the extracted methods for onSaved function.

Reply:
is this needed? isn't this redundant?

Reply:
I tried omitting the onSaved function, the edited product does not reflect.